### DISCUSSION TITLE
AIR Native Extension Issue

### DISCUSSION THREAD

**Discussion by freelanceralphaomega** [airsdk/Adobe-Runtime-Support#903] [General]
I built an AIR native extension with following setup:
1. Visual studio 2005 for building my dll
2. Flash CS6 IDE with AIR SDK 3.2 to create SWC from my AS3 source
3. The namespace version of my ANE is 33.1
4. ADT (AIR SDK 33.1)
I am testing the ANE with FlashDevelop 5.3.3.1 and the ANE works as expected on a machine that I use for development. However, when I test the same ANE in FlashDevelop on a different machine, the AIR can't find a necessary method in the native side implementation (Error code #3500). I am using the same version of FlashDevelop with same AIR SDK (33.1) in both machines.
It seems that for some reason AIR could not load the dll in second machine (but does so in my development machine), and I am not sure why. Any help would be highly appreciated.

Thank you.

**Comment by ajwfrost** [comment]
If you're developing using Visual Studio, then it might be that your DLL has a dependency upon the Visual C++ redistributable package...
2005 is very old, I don't think Microsoft still provide them, although you may have them already on the development computer.
See https://support.microsoft.com/en-us/topic/the-latest-supported-visual-c-downloads-2647da03-1eea-4433-9aff-95f26a218cc0
and look for a file "redist.txt" in your Visual Studio installation folder, or search for "vcredist_x86.exe" etc... 

But to tell for sure if this is a problem with a DLL dependency: try dropping the DLL into the Dependency Walker:
https://dependencywalker.com/
It tends to take ages to analyse a file, but is then very good at seeing if there are missing dependencies/imports (once you can see past the false positives...)

**Reply by freelanceralphaomega** [reply]
Thank you ajwfrost, that was indeed the problem. I found the Visual C++ redistributable package for VS 2005, built my dll in Release mode and it resolved my issue.
